module/VuFind/src/VuFind/Auth/Shibboleth.php

```diff
@@ -228,18 +250,9 @@ public function getSessionInitiator($target)
      */
     public function isExpired()
     {
-        $config = $this->getConfig();
-        if (isset($config->Shibboleth->username)
-            && isset($config->Shibboleth->logout)
-        ) {
-            // It would be more proper to call getServer on a Laminas request
-            // object... except that the request object doesn't exist yet when
-            // this routine gets called.
-            $username = isset($_SERVER[$config->Shibboleth->username])
-                ? $_SERVER[$config->Shibboleth->username] : null;
-            return empty($username);
-        }
-        return false;
+        $proxy = $this->getConfig()->Shibboleth->proxy ?? false;
+        return ($proxy) ? isset($_SERVER[$this->normalize(self::SHIB_SESSION_ID)])
```

Comment:
Are you sure these changes are correct/appropriate? I see a few potential issues:

1.) This method is checking to see if the login is expired, but it's checking to see if a session ID is set. Does the logic need to be reversed to check to see if things are NOT set?

2.) It would be better to avoid using PHP superglobals. Can you inject the framework's server object through the constructor so you have access to it here? (You can retrieve it with `$container->get('Request')->getServer()` in the factory). Can you use `getenv()` instead of `$_ENV`?

Reply:
I fixed the code and added documentation that was removed by mistake.